### ISSUE TITLE
failed to load data from dynamodb

### ISSUE THREAD

**Issue by rajeshwer** [mneil/opa-dynamodb#13]
hi

i am looking to use dynamodb for below data

```
{
    "allowed_amis": {
        "Amazon": {
            "value": "id"
        }
    }
}

```
below is my policy
https://play.openpolicyagent.org/p/9GWfNsFAxe

when i try to load data from dynamodb i am getting below error

```
curl -X POST http://127.0.0.1:8181/v1/data/amiid --data-binary '{ "input": {"path": "allowed_amis.Amazon.value","value": "id"}}' |jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   469  100   406  100    63   1691    262 --:--:-- --:--:-- --:--:--  1954
{
  "code": "internal_error",
  "message": "error(s) occurred while evaluating query",
  "errors": [
    {
      "code": "eval_builtin_error",
      "message": "dynamodb.policy: ValidationException: Query key condition not supported\n\tstatus code: 400, request id: 602b4674-7749-4463-8955-2f18b49ed817",
      "location": {
        "file": "amiid",
        "row": 9,
        "col": 12
      }
    }
  ]
}
```

**data in dynamodb**

```
aws dynamodb scan --table-name OpaDynamoDB --endpoint-url http://localhost:8000 
{
    "Items": [
        {
            "allowed_amis.Amazon.value": {
                "S": "id"
            },
            "pk": {
                "S": "foo/bar"
            }
        }
    ],
    "Count": 1,
    "ScannedCount": 1,
    "ConsumedCapacity": null
}

```
i am using below policy

package amiid
  

default allowresult = "invalid ami image"

```
allowresult {
   # Lookup ami from data
   path := dynamodb.policy("foo/bar", input.path)
   input.value = path.value
}
```

can you provide a working example on how to use this 
Thanks

**Comment by mneil** [comment]
Thanks for reporting. Internally it expects that you have a hash key and range key for your index.  Is allowed_amis your range key? I'll load this up this weekend and try it myself but my guess is incompatible keys. 

I didn't document that requirement. I'll create a separate issue.

**Comment by rajeshwer** [comment]
Sure thanks

**Comment by mneil** [comment]
Ok, there was actually an issue which has been addressed in #16 . Tests are failing there currently because of some missing region settings in the automated tests but pass locally. I'll get those resolved later and merged in.

I have created https://mneil.github.io/opa-dynamodb with a quickstart that shows the schema requirements and examples https://mneil.github.io/opa-dynamodb/#/quickstart.

I've also published to docker hub https://hub.docker.com/r/mneil/opa-dynamodb which has the latest fixes in it. Try that out and check out the docs and let me know if that resolves the issue for you.
